### PR TITLE
Corners

### PR DESCRIPTION
**Pull request by tuttarealstep**
corners

### CODE REVIEW

File: src/objects/room/StairCorner.ts [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+        cornerTwo.height = 8;
+        cornerTwo.zIndex = 0;
+
+        return [tile, borderLeft, cornerOne, cornerTwo]
+    } else {
+        return [tile, borderLeft]
+    }
+}
+
+export function createStairBoxDirection2({
```
Comment (tuttarealstep) [inline src/objects/room/StairCorner.ts:343]: Need to delete this, from stair class

Comment (tuttarealstep) [reply src/objects/room/StairCorner.ts:343]: createStairBoxDirection2

**Comment by jankuss** [comment]
Thanks for the PR! :+1:

While testing I ran into the following issue:
image

If I had to guess, it is because the logic for determining corners is flawed.

I think the rules for corner stairs are:
- There is a stair at the above (Position: y - 1) tile at the same height with direction 2
- There is a stair to the left (Position: x - 1) at the same height with direction 0

File: src/util/getTileInfo.ts [hunk 2]
```diff
@@ -81,36 +87,50 @@ function getStairs(tiles: TileType[][], x: number, y: number) {
...
     const diff = Number(topType) - Number(type);
 
     if (diff === 1) {
-      const destinationStairs = getStairs(
-        tiles,
-        x + offsets.top.x,
-        y + offsets.top.y
-      );
-
-      if (destinationStairs == null || destinationStairs.direction === 0) {
```
Comment (tuttarealstep) [inline src/util/getTileInfo.ts:95]: The problem with your screen is from this check. Without this as u can see all seem okay.

fix_stairs

fix_stairs2

**Pull request closed without merging**